diff --git a/core/modules/workflows/src/Plugin/WorkflowTypeBase.php b/core/modules/workflows/src/Plugin/WorkflowTypeBase.php
index 6045484244..0957caf89c 100644
--- a/core/modules/workflows/src/Plugin/WorkflowTypeBase.php
+++ b/core/modules/workflows/src/Plugin/WorkflowTypeBase.php
@@ -151,6 +151,12 @@ public function getStates($state_ids = NULL) {
    * {@inheritdoc}
    */
   public function getState($state_id) {
+    // awfull hack to get first state from config in case empty initial state
+    // @see https://www.drupal.org/project/drupal/issues/2901459#comment-12568526 which should fix this
+    if (empty($state_id)) {
+      $state_id = key($this->configuration['states']);
+    }
+
     if (!isset($this->configuration['states'][$state_id])) {
       throw new \InvalidArgumentException("The state '$state_id' does not exist in workflow.'");
     }
